src/tls.c

```diff
@@ -422,10 +422,17 @@ static char *ssl_printname(X509_NAME *name)
 
   /* X509_NAME_oneline() is easier and shorter, but is deprecated and
      the manual discourages it's usage, so let's not be lazy ;) */
-  X509_NAME_print_ex(bio, name, 0, XN_FLAG_ONELINE & ~XN_FLAG_SPC_EQ);
-  len = BIO_get_mem_data(bio, &data) + 1;
-  buf = nmalloc(len);
-  strlcpy(buf, data, len);
+  if (X509_NAME_print_ex(bio, name, 0, XN_FLAG_ONELINE & ~XN_FLAG_SPC_EQ)) {
+    len = BIO_get_mem_data(bio, &data);
```

Comment:
According to docs, this returns a `long` so let's change it while we're at it. It also seems to be signed and though it's unclear if it can get negative, please add a check below.
Same comment for the same codeblocks lower.